File: 349.intersection_of_two_arrays/intersection.js
/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number[]}
 */
var intersection = function(nums1, nums2) {
    var result = [];
    var shortArray = nums1.length > nums2.length ? nums2 : nums1;
    var longArray = nums1.length > nums2.length ? nums1 : nums2;
    
    for(var i = 0; i < shortArray.length; i++){
        var value = shortArray[i];
        if(longArray.indexOf(value) !== -1 && result.indexOf(value) === -1){
            result.push(value);
        }
    }
    
    return result;
};

File: 349.intersection_of_two_arrays/intersection.py
class Solution(object):
    def intersection(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: List[int]
        """
        shorter = nums1 if len(nums1) <= len(nums2) else nums2
        longer = nums2 if len(nums2) >= len(nums1) else nums1
        s = set()
        res = set()
        for i in longer:
            s.add(i)
        for j in shorter:
            if j in s:
                res.add(j)
        return list(res)
        
        # one line
        # return list(set(nums1) & set(nums2))